refactor: 예외 처리 개선 및 불필요한 코드 제거

cje206 opened the pull request:
## #️⃣ 연관된 이슈

> #74 

## #️⃣ 작업 내용

> 예외 처리 개선 및 불필요한 코드 제거

## #️⃣ 테스트 결과

> 빌드 성공
> 상품 조회 성공 (es 있는 경우 조회 가능, es 꺼져있는 경우 PRODUCT-503)

## #️⃣ 변경 사항 체크리스트

- [x] 코드에 영향이 있는 모든 부분에 대한 테스트를 작성하고 실행했나요?
- [ ] 문서를 작성하거나 수정했나요? (필요한 경우)
- [x] 코드 컨벤션에 따라 코드를 작성했나요?
- [x] 본 PR에서 발생할 수 있는 모든 의존성 문제가 해결되었나요?

## #️⃣ 스크린샷 (선택)

> 

## #️⃣ 리뷰 요구사항 (선택)

> product-service가 정상 빌드 되는지 확인 부탁드립니다.
> get 조회 가능한지 확인 부탁드립니다.

## 📎 참고 자료 (선택)

>

metroz1 committed:
refactor: authentication 활용하는 방식으로 변경

metroz1 committed:
refactor: Kafka 이벤트 구조 변경

cje206 committed:
refactor: 예외 처리 개선 및 불필요한 코드 제거

metroz1 committed:
refactor: 서비스 의존성 수정 및 렌탈 서비스 리팩토링

metroz1 commented on modi/product-service/src/main/java/com/coc/modi/product/product/application/ProductService.java:
```diff
@@ -8,180 +8,194 @@
+	// 3-4. 상품 등록
+	@Transactional
+	public ProductResponse createProduct(Long memberId, ProductCommand command) {
+		
```
컨트롤러에서 Request -> Command로 변경해서 넘길 때 memberId도 함께 command에 넣는 건 어떨까용

cje206 replied:
좋은 아이디어네요!

metroz1 commented on modi/product-service/src/main/java/com/coc/modi/product/product/application/ProductService.java:
```diff
@@ -8,180 +8,194 @@
+				ProductCategory.from(command.category()));
+		
+		// 이미지 추가
+		product.addImages(command.imageUrls());
```
Product.create 메소드에서 image를 추가하는 건 어떤가요?

cje206 replied:
넵 좋은 아이디어입니다!

carsumin committed:
Merge pull request #81 from prgrms-be-adv-devcourse/refactor/rental-service

Refactor/rental service

metroz1 commented on modi/product-service/src/main/java/com/coc/modi/product/product/infrastructure/ProductImageRepositoryAdapter.java:
```diff
@@ -2,20 +2,21 @@
 
 import java.util.Optional;
 
 @Repository
```
@RequiredArgsConstructor 로 DI 하는 부분 통일 부탁드립니다.

cje206 replied:
넵 반영하겠습니다.

metroz1 commented on modi/product-service/src/main/java/com/coc/modi/product/product/presentation/ProductController.java:
```diff
@@ -12,92 +12,95 @@
+	// 상품 등록
+	@PostMapping
+	public ResponseEntity<ApiResponse<ProductResponse>> createProduct(Authentication authentication,
+																	  @RequestBody ProductRequest request) {
```
ProductRequest도 좋지만 ProductCreateRequest 처럼 조금 더 확실하게 네이밍을 하는 건 어떨까용

cje206 replied:
넵 반영하겠습니다

metroz1 commented on modi/product-service/src/main/java/com/coc/modi/product/product/presentation/ProductController.java:
```diff
@@ -12,92 +12,95 @@
+		
+		Long memberId = (Long)authentication.getPrincipal();
+		
+		ProductCommand command = new ProductCommand(
```
request.toCommand() 메소드로 커맨드 만드는 방식으로 통일 하는 건 어떨까용

cje206 replied:
넵 좋은 아이디어입니다

metroz1 commented on modi/product-service/src/main/java/com/coc/modi/product/product/presentation/ProductController.java:
```diff
@@ -12,92 +12,95 @@
+	
+	// 상품 숨김
+	@PatchMapping("/{productId}")
+	public ResponseEntity<ApiResponse<?>> disableProduct(Authentication authentication,
```
ApiResponse<?>은 보통 어떤 타입을 리턴할지 모르거나 혹은 상황에 따라 반환할 타입이 달라지거나 하는 경우에 사용하는 것이 좋지 않을까 합니다! delete 등의 액션으로 반환할 바디가 없다는게 명확하다면 <Void>를 사용하는게 어떨까용

cje206 replied:
넵 확인했습니다

metroz1 reviewed (commented):
코멘트 남긴 부분 확인 부탁드립니다!

cje206 committed:
refactor: product-service 수정 요청 사항 반영

- command record에 memberId 및 toCommand 추가
- response, request, repository, service 등의 이름을 명확하게 명시
- controller, request에서 null값 검증
- annotation 순서 정리
- 코드 포멧 맞추기
- 메서드명 상황에 맞게 직관적으로 수정
- 상태 변경 서비스 분리 및 활성화 api 추가
- ResponseEntity 정리
- Product.create메서드에서 addImage 처리
- null값 검증 controller와 request에서 처리
- dto 의미가 명확하게 리네이밍

cje206 committed:
Merge branch 'dev-rental' of https://github.com/prgrms-be-adv-devcourse/beadv2_2_COC_BE into refactor/product

metroz1 reviewed (approved):
변경된 부분 확인 했습니다.

cje206 committed:
refactor: elasticsearch 오류 에러 수정, sellerId 조회 시 예외 처리 추가
